autoemulate/experimental/emulators/randomforest.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.utils.validation import check_X_y
+
+from autoemulate.experimental.emulators.base import SklearnBackend
+from autoemulate.experimental.types import TensorLike
+
+
+class RandomForest(SklearnBackend):
+    """Random forest Emulator.
+
+    Implements Random Forests regression from scikit-learn.
+    """
+
+    def __init__(  # noqa: PLR0913 allow too many arguments since all currently required
+        self,
+        x: TensorLike,
+        y: TensorLike | None,
+        n_estimators=100,
+        criterion="squared_error",
+        max_depth=None,
+        min_samples_split=2,
+        min_samples_leaf=1,
+        max_features=1.0,
+        bootstrap=True,
+        oob_score=False,
+        max_samples=None,
+        random_state=None,
+    ):
+        """Initializes a RandomForest object."""
+        _, _ = x, y  # ignore unused arguments
+        self.n_estimators = n_estimators
+        self.criterion = criterion
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.max_features = max_features
+        self.bootstrap = bootstrap
+        self.oob_score = oob_score
+        self.max_samples = max_samples
+        self.random_state = random_state
+        self.model = RandomForestRegressor(
+            n_estimators=self.n_estimators,
+            max_depth=self.max_depth,
+            min_samples_split=self.min_samples_split,
+            min_samples_leaf=self.min_samples_leaf,
+            max_features=self.max_features,
+            bootstrap=self.bootstrap,
+            random_state=self.random_state,
+        )
+
+    def fit(self, x: TensorLike, y: TensorLike | None):
+        """Fits the emulator to the data."""
+        x, y = self._convert_to_numpy(x, y)
+        self.n_features_in_ = x.shape[1]
+
+        # y = y.ravel()  # Ensure y is 1-dimensional
```

Comment:
Doing this would break the 2d test case, however not doing it results in the following warning even on the 1D test (caused by our dimensionality handling in the `_convert_...` functions:

```
A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel()
```

This is probably fine though